server/service/user-service.js

```diff
@@ -0,0 +1,119 @@
+const db = require('../db');
+const bcrypt = require('bcrypt');
+const uuid = require('uuid');
+const mailService = require('./mail-service');
+const tokenService = require('./token-service');
+const UserDto = require('../dtos/user-dto');
+const ApiError = require('../exeptions/api-error');
+
+class UserService {
+  async registration(email, password, name, last_name, role) {
+    const candidate = await db.query('SELECT * FROM users WHERE email = $1', [
+      email,
+    ]);
+
+    if (candidate.rows[0]) {
+      throw ApiError.BadRequest(`User with email ${email} already exists`);
+    }
+
+    const hashPassword = await bcrypt.hash(password, 3);
+    const activationLink = uuid.v4();
+
+    const user = await db.query(
+      'INSERT INTO users (email, password, name, last_name, role, activationLink) values ($1, $2, $3, $4, $5, $6) RETURNING *',
```

Comment:
were you going to use an ORM for this?

Reply:
I've built this with raw SQL before we discuss whether we will build it with ORM or not.
Now I'll refactor code and I'll integrate ORM.

Reply:
gotcha no worries. leave it as is. and make a separate branch/ticket to complete that refactor separately. so that it's easier to view instead of this massive PR.

Reply:
Got it! Will do when we merge the branches.